libs/langchain-community/src/indexes/postgres.ts

```diff
@@ -20,9 +21,19 @@ export class PostgresRecordManager implements RecordManagerInterface {
   namespace: string;
 
   constructor(namespace: string, config: PostgresRecordManagerOptions) {
-    const { postgresConnectionOptions, tableName } = config;
+    const { postgresConnectionOptions, pool, tableName } = config;
     this.namespace = namespace;
-    this.pool = new pg.Pool(postgresConnectionOptions);
+    if (pool instanceof Pool) {
+      this.pool = pool
+    } else {
+      try {
+        this.pool = new pg.Pool(postgresConnectionOptions);
+      } catch (err: unknown) {
```

Comment:
pool constructor does not throw when config is undefined/invalid as far as I can tell. checking that pool and config aren't both undefined should work